src/common.ts

```diff
@@ -68,7 +105,49 @@ export interface Transferable {
 }
 
 export interface Task {
-  readonly [kQueueOptions] : object | null;
+  readonly [kQueueOptions] : {} | null;
+  filename: string;
+  name: string;
+  taskId: number;
+}
+
+export abstract class AsynchronouslyCreatedResource {
+  onreadyListeners : (() => void)[] | null = [];
+
+  markAsReady () : void {
+    const listeners = this.onreadyListeners;
+    assert(listeners !== null);
+    this.onreadyListeners = null;
+    for (const listener of listeners) {
+      listener();
+    }
+  }
+
+  isReady () : boolean {
+    return this.onreadyListeners === null;
+  }
+
+  onReady (fn : () => void) {
+    if (this.isReady()) {
+      fn(); // Zalgo is okay here.
```

Comment:
Is it? :)

Reply:
On that, I'm not sure; was in this way so I just port it. But it might be worth it to see if zalgo is really ok there 🤔

Reply:
just do a queueMicrotask to be sure

Reply:
Believe it or not, it seems to matter 😅 
If trying to offload it to the micro task or before the next tick, it hangs; it seems we just add stacks to the same tick and promises don't allow us to move forward in the loop; causing Piscina to hang.

Interesting gotcha